moderation/register.py

```diff
@@ -142,9 +143,9 @@ def pre_save_handler(self, sender, instance, **kwargs):
             moderated_obj = self._get_or_create_moderated_object(instance,
                                                                  unchanged_obj,
                                                                  moderator)
-            if moderated_obj.moderation_status != \
-               MODERATION_STATUS_APPROVED and \
-               not moderator.bypass_moderation_after_approval:
+            if not (moderated_obj.moderation_status ==
+                    MODERATION_STATUS_APPROVED and
+                    moderator.bypass_moderation_after_approval):
```

Comment:
@blag: I honestly don't remember anything about this at all now, it's been a while. Please use my changes in your patch if it still applies, with any fixes you think are needed.